src/formatter.rs

```diff
@@ -353,10 +353,26 @@ fn color_for_name(color_name: &str) -> Option<Color> {
 
 impl<W: Write> Write for ColorFormatter<W> {
     fn write(&mut self, data: &[u8]) -> Result<usize, Error> {
-        if self.has_pending_style_change {
-            self.write_new_style()?;
+        for line in data.split_inclusive(|b| *b == b'\n') {
+            if line.ends_with(b"\n") {
+                if self.has_pending_style_change {
+                    self.write_new_style()?;
+                }
+                self.output.write_all(&line[..line.len() - 1])?;
+                let labels = self.labels.clone();
+                self.labels.clear();
+                self.write_new_style()?;
```

Comment:
`mem::take()` or `mem::replace()` can be used to temporarily reset labels, but I think it would be better to pass `labels` as an argument.

Reply:
I went with `mem::tale()` and `mem::replace()` because `write_new_style()` mutably borrows `self`, so we would need to pass a clone of the labels otherwise. Let me know if there was a way that I just missed.

Reply:
Right. Maybe we would have to break things further to pass labels as argument.

Nit: `mem::replace()` without using old value is just an assignment.

Reply:
D'oh. That's why it wanted me to use the result. Fixed :)